Client/Frontend/Sync/SyncViewController.swift

```diff
@@ -43,4 +49,12 @@ class SyncViewController: UIViewController {
         
         code()
     }
+    
+    @objc func didLeaveSyncGroup() {
+        DispatchQueue.main.async { [weak self] in
+            if let settingsVC = self?.navigationController?.childViewControllers.first {
```

Comment:
lgtm, only critique is changing name, instead of `settingsVC` re-name to something like `rootVC` as it may not always be a settingsvc (although currently it always is).